### PR TITLE
re-add keep page, update css to handle comments section

### PR DESCRIPTION
**Pull request by camhashemi**
@cvializ @DerekBurch

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/keep/keepCardAttribution.tpl.html [hunk 1]
```diff
@@ -65,8 +65,10 @@
     <span ng-if="showSaved && keep.summary.hasContent"><a
       ng-href="https://www.kifi.com/k/cached?id={{keep.id}}">Saved</a></span>
     <span>
-      <time ng-attr-datetime="{{::keep.createdAt}}" am-time-ago="::keep.createdAt"
-            title="{{::keep.createdAt|localTime}}" ng-if="::keep.createdAt"></time>
+      <a class="kf-keep-card-attribution-page-link" ng-href="{{::keep.path}}">
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCardAttribution.tpl.html]: Could we make this admin only for now?

**Comment by camhashemi** [comment]
this got removed during merge resolution, which is probably for the best (may even have been intentional) because the new keep card changes broke things

File: kifi-backend/modules/shoebox/angular/src/keep/keepCardAttribution.tpl.html (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCardAttribution.tpl.html]: yup!

**Comment by cvializ** [comment]
lgtm!

File: kifi-backend/modules/common/app/com/keepit/model/Keep.scala [hunk 2]
```diff
@@ -91,7 +91,7 @@ case class Keep(
       ).forall(b => b))
   }
 
-  def titlePathString = this.title.map(_.trim.replaceAll(" ", "-")).getOrElse(this.url.trim.replaceAll("[./]", "-")).take(40)
+  def titlePathString = this.title.map(_.trim.replaceAll(" ", "-")).getOrElse(this.url.trim.replaceAll("^https?://", "").replaceAll("\\?.*", "").replaceAll("[./]", "-")).take(40)
```
Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/model/Keep.scala:94]: @andrewconner

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala [hunk 3]
```diff
@@ -148,12 +148,17 @@ class KeepDecoratorImpl @Inject() (
               augmentationInfoForKeep.libraries.collect { case (libraryId, contributorId, keptAt) if doShowLibrary(libraryId) => (BasicLibraryWithKeptAt(idToBasicLibrary(libraryId), keptAt), idToBasicUser(contributorId)) }
             }
 
+            val bestEffortPath = (keep.title, pageInfoForKeep.title) match {
+              case (None, Some(title)) => keep.copy(title = Some(title)).path.relative
+              case _ => keep.path.relative
+            }
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala:154]: @andrewconner

File: kifi-backend/modules/shoebox/angular/src/keep/keepCardAttribution.tpl.html [hunk 4]
```diff
@@ -60,13 +60,16 @@
...
+    <span ng-if="isAdmin && keep.summary.hasContent">·</span>
+    <span ng-if="isAdmin && keep.summary.hasContent"><a
+      ng-href="https://www.kifi.com/k/cached?id={{keep.id}}" target="_blank">Saved</a></span>
     <span ng-if="::keep.createdAt">·</span>
-    <span ng-if="showSaved && keep.summary.hasContent"><a
-      ng-href="https://www.kifi.com/k/cached?id={{keep.id}}">Saved</a></span>
     <span>
-      <time ng-attr-datetime="{{::keep.createdAt}}" am-time-ago="::keep.createdAt"
-            title="{{::keep.createdAt|localTime}}" ng-if="::keep.createdAt"></time>
+      <a class="kf-keep-card-attribution-page-link" ng-href="{{ isAdmin ? keep.path : '' }}">
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCardAttribution.tpl.html]: @cvializ added `isAdmin` to this directive

File: kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html [hunk 5]
```diff
@@ -8,7 +8,8 @@
           <div kf-keep-card-attribution
                keep="keep"
                show-library-attribution="::(currentPageOrigin !== 'libraryPage' || showOriginLibrary)"
-               is-first-item="::isFirstItem"></div>
+               is-first-item="::isFirstItem"
+               is-admin="::isAdmin"></div>
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html]: How about just check inside `kf-keep-card-attribution` for isAdmin? One less thing to pass in.

File: kifi-backend/modules/shoebox/angular/src/keep/keepCardAttribution.tpl.html (hunk 4 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCardAttribution.tpl.html]: Only include `<a>` if we're going to link it (don't have an empty `href`)

**Comment by andrewconner** [comment]
Minor comments, rest :+1:

File: kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html (hunk 5 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html]: sounds good. I wasn't sure what the best protocol is.. that way repeats computation of the admin exp and injects `profileService` for something small, this way you have to pass things around a lot.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html]: Extra injection is fine. Compare it to a method in a dependency-injected class. You pass parameters to the method, and inject dependencies. Usually, when it's a static dependency (ie, we always use it the same way), inject away.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala (hunk 3 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala:154]: :+1:

**Pull request merged**